bindings/python/pymongoarrow/api.py

```diff
@@ -233,3 +239,42 @@ def aggregate_numpy_all(collection, pipeline, *, schema, **kwargs):
     return _arrow_to_numpy(
         aggregate_arrow_all(collection, pipeline, schema=schema, **kwargs), schema
     )
+
+
+def write(collection, tabular, mode="insert"):
+    validate_schema(tabular.schema)
+    cur_batch = []
+    cur_size = 0
+    cur_offset = 0
+    from collections import Counter
+
+    results = Counter({})
+
+    def transform_bwe(bwe, offset):
+        bwe["nInserted"] += cur_offset
+        for i in bwe["writeErrors"]:
+            i["index"] += offset
+        return bwe
+
+    for row in tabular.to_pylist():
+        if cur_size <= MAX_BSON_SIZE and len(cur_batch) <= MAX_WRITE_BATCH_SIZE:
+            cur_batch.append(row)
+            cur_size += len(encode(row))
+        else:
+            try:
+                collection.insert_many(cur_batch)
+            except BulkWriteError as bwe:
```

Comment:
How should we handle other exceptions that come from pymongo, like ConnectionError, NotPrimaryError, ServerSelectionTimeoutError, etc...?